types/timestamp.go

```diff
@@ -31,3 +31,17 @@ func (ts TimestampSlice) Less(i, j int) bool {
 func (ts TimestampSlice) Swap(i, j int) {
 	ts[i], ts[j] = ts[j], ts[i]
 }
+
+// Clock allows clients to retrieve the current time.
+type Clock interface {
+	Now() Timestamp
+}
+
+// StdClock is an implementation of Clock that retrieves the current time using
+// the system time.
+type StdClock struct{}
+
+// Now retrieves the current timestamp.
+func (c StdClock) Now() Timestamp {
+	return Timestamp(time.Now().Unix())
+}
```

Comment:
You make a clock, but you don't use it in any tests.

Also, we should settle on a file layout. @lukechampine @mtlynch , this is my proposal:

```
package xxx

import (
    "stdlib"

    "external packages"

    "internal packages"
)

const (
    // All constants, grouped sensibly
)

var (
    // things that should be consts, but aren't type-compatible

    // errors

    // other stuff, grouped sensibly
)

type (
    // aliases

    // dependency/mocking interfaces + structs

    // types that implement the mocking interfaces (like `StdClock`)
)

// any functions on the mocking types (like `keyDeriver`)

// rest of file
```

You put all of the mocking stuff near the top.